custom/sensors/TalonEncoder.java

```diff
@@ -0,0 +1,89 @@
+package org.usfirst.frc4904.standard.custom.sensors;
+
+
+import edu.wpi.first.wpilibj.CANTalon;
+import edu.wpi.first.wpilibj.PIDSourceType;
+
+public class TalonEncoder implements CustomEncoder {
+	protected final CANTalon talon;
+	private PIDSourceType pidSource;
+	private double distancePerPulse;
+	private boolean reverseDirection;
+	private boolean rateMode;
+	
+	public TalonEncoder(String name, CANTalon talon, boolean reverseDirection, double distancePerPulse, boolean rateMode) {
+		this.talon = talon;
+		this.reverseDirection = reverseDirection;
+		this.distancePerPulse = distancePerPulse;
+		this.rateMode = rateMode;
+		setPIDSourceType(PIDSourceType.kDisplacement);
+	}
+	
+	@Override
+	public void setPIDSourceType(PIDSourceType pidSource) {
+		this.pidSource = pidSource;
+	}
+	
+	@Override
+	public PIDSourceType getPIDSourceType() {
+		return pidSource;
+	}
+	
+	@Override
+	public double pidGet() {
```

Comment:
I think we should piggyback off of PIDSourceType, PIDSource, and PIDOutput for our PID framework - the framework makes sense save for PIDController. CustomPID, calling pidGet, can also figure out whether its source is returning rate or displacement, and then we set the output type as well as the input type (this is what PIDController is missing.)